app/assets/javascripts/SourceCodeGlower/SourceCodeLine.js

```diff
@@ -22,26 +24,201 @@ function SourceCodeLine(line_node) {
   this.is_observing      = false;
 }
 
-// Increase a Source Code Line's glow depth
-SourceCodeLine.prototype.glow = function() {
-  // Increase the depth
+// Increase a Source Code Line's glow depth, add listeners
+// Splits nodes at start and end offsets if needed
+SourceCodeLine.prototype.glow = function(annotation_id, start, end,
+  hover_on_function, hover_off_function) {
+  // Increase the depth and turn on design mode for highlighting
   this.incGlowDepth(1);
 
-  // Add the appropriate glow class
-  this.getLineNode().addClass('source_code_glowing_' + this.getGlowDepth());
+  // Save line node and get all text nodes in the line
+  var node = this.getLineNode();
+  var textNodes = this.getAllTextNodes(node);
+
+  // Loop prep, for tracking nodes and character counts
+  var start_node = null;
+  var start_node_offset = 0;
+  var end_node = null;
+  var end_node_offset = 0;
+  var foundStart = false;
+  var currentCharCount = 0;
+
+  // Loop over nodes, add length of passed nodes, save start and end then break
+  for (var i = 0; i < textNodes.length; i++) {
+    // End char offset of the current text node
+    var endCharCount = currentCharCount + textNodes[i].length;
+
+    // If start is between current count and end of the current node, save it
+    if (!foundStart && start >= currentCharCount && start <= endCharCount) {
+      foundStart = true;
+
+      // Will need to split the node if the glow starts inside of it
+      if (start != endCharCount) {
+        // Save reference to span node, create span wrapper if blank space
+        if ( textNodes[i].parentNode.parentNode == node){
```

Comment:
The whitespace here is odd - no space after `(`, space before `{`.

Also, you should use `===` here